apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -334,14 +383,19 @@ public SpannerConnectionStringBuilder(string connectionString, ChannelCredential
         public SpannerConnectionStringBuilder(string connectionString, ChannelCredentials credentials, SessionPoolManager sessionPoolManager)
         {
             ConnectionString = GaxPreconditions.CheckNotNull(connectionString, nameof(connectionString));
-            CredentialOverride = credentials;
+            // If SPANNER_EMULATOR_HOST env var is set, ignore any provided credentials and use insecure credentials.
+            ParseEmulatorHostAndPort();
+            CredentialOverride = string.IsNullOrEmpty(_emulatorHost) ? credentials : ChannelCredentials.Insecure;
```

Comment:
I believe this is wrong. If the Host and the Port were already specified in the connection string, the emulator will be ignored even if it is set (that's how Host and Port are implemented, the specified values take precedence over the emulator values).
In here you cannot override the credentials if the emulator is set, you can override them if the emulator is set and Host and Port were not set in the connection string.

Reply:
No longer relevant since the implementation has changed.